### PR TITLE
CIS-1457 Handle notification.channel_deleted event

### PR DESCRIPTION
**Pull request by tbarbugli**
Handle `notification.channel_deleted` event so that we update the Channel List correctly

### CODE REVIEW

File: Sources/StreamChat/WebSocketClient/Events/NotificationEvents.swift [hunk 1]
```diff
@@ -416,3 +416,35 @@ class NotificationInviteRejectedEventDTO: EventDTO {
         )
     }
 }
+
+/// Triggered when the current user rejects an invite to a channel.
+public struct NotificationChannelDeletedEvent: ChannelSpecificEvent {
+    let cid: ChannelId
+    let channel: ChatChannel
+    let createdAt: Date
```
Comment (polqf) [inline Sources/StreamChat/WebSocketClient/Events/NotificationEvents.swift:424]: Is this createdAt property the date in which the channel got deleted? if that's the case, can we document it?

File: Sources/StreamChat/WebSocketClient/Events/NotificationEvents.swift [hunk 2]
```diff
@@ -416,3 +416,35 @@ class NotificationInviteRejectedEventDTO: EventDTO {
...
+    let payload: EventPayload
+}
+
+class NotificationChannelDeletedEventDTO: EventDTO {
+    let cid: ChannelId
+    let channel: ChannelDetailPayload
+    let createdAt: Date
```
Comment (polqf) [inline Sources/StreamChat/WebSocketClient/Events/NotificationEvents.swift:431]: Same here

**Review by nuno-vieira — APPROVED** [review]
LGTM! ✅

File: Sources/StreamChat/WebSocketClient/Events/NotificationEvents.swift [hunk 3]
```diff
@@ -416,3 +416,38 @@ class NotificationInviteRejectedEventDTO: EventDTO {
...
+/// Triggered when a channel is deleted, this event is delivered to all channel members
+public struct NotificationChannelDeletedEvent: ChannelSpecificEvent {
+    /// The cid of the deleted channel
+    let cid: ChannelId
+
+    /// The channel that was deleted
+    let channel: ChatChannel
+
+    /// The event timestamp.
+    let createdAt: Date
```
Comment (nuno-vieira) [inline Sources/StreamChat/WebSocketClient/Events/NotificationEvents.swift:429]: Shouldn't these properties be public?

**Review by nuno-vieira — CHANGES_REQUESTED** [review]
Requesting changes since I believe we are missing `public` on some props

**Pull request merged**